src/app/pages/storage/snapshots/snapshot-table/snapshot-list.component.ts

```diff
@@ -0,0 +1,327 @@
+  }
+
+  expandRow(row: SnapshotListRow): void {
+    this.expandedRow = this.expandedRow === row ? null : row;
+    this.cdr.markForCheck();
+  }
+
+  toggleExtraColumnsDialog(): void {
+    this.showExtraColumns$.next(!this.showExtraColumns$.value);
+    window.localStorage.setItem('snapshotXtraCols', this.showExtraColumns$.value.toString());
+    if (this.showExtraColumns$.value) {
+      this.store$.dispatch(loadSnapshots({ extra: true }));
+    }
+    this.getSnapshots();
+    this.selection.clear();
+    const slideToggleControl = this.toolbarConfig.controls.find((control) => control.name === 'extra-columns');
+    slideToggleControl.confirmOptions = this.getConfirmOptions();
+    this.cdr.markForCheck();
+  }
+
+  onAction(event: SnapshotListEvent): void {
+    switch (event.action) {
+      case 'rollback':
+        this.doRollback(event.row);
+        break;
+      case 'clone':
+        this.doClone(event.row);
+        break;
+      case 'delete':
+        this.doDelete(event.row);
+        break;
+    }
+  }
+
+  setupToolbar(): void {
+    this.settingsEvent$.pipe(
+      untilDestroyed(this),
+    ).subscribe((event: CoreEvent) => {
+      switch (event.data.event_control) {
+        case 'filter':
+          this.filterString = event.data.filter;
+          this.dataSource.filter = event.data.filter;
+          break;
+        case 'add':
+          this.doAdd();
+          break;
+        case 'extra-columns':
+          this.toggleExtraColumnsDialog();
+          break;
+        default:
+          break;
+      }
+    });
+
+    const controls: ControlConfig[] = [
+      {
+        name: 'extra-columns',
+        type: 'slide-toggle',
+        label: this.translate.instant('Show extra columns'),
+        value: this.showExtraColumns$.value,
+        confirmOptions: this.getConfirmOptions(),
+      },
+      {
+        name: 'filter',
+        type: 'input',
+        value: this.filterString,
+        placeholder: this.translate.instant('Search'),
+      },
+      {
+        name: 'add',
+        type: 'button',
+        label: this.translate.instant('Add'),
+        color: 'primary',
+        ixAutoIdentifier: 'Snapshots_ADD',
+      },
+    ];
+
+    const toolbarConfig: ToolbarConfig = {
+      target: this.settingsEvent$,
+      controls,
+    };
+    const settingsConfig = {
+      actionType: EntityToolbarComponent,
+      actionConfig: toolbarConfig,
+    };
+
+    this.toolbarConfig = toolbarConfig;
+    this.core.emit({ name: 'GlobalActions', data: settingsConfig, sender: this });
+  }
+
+  selectAll(): void {
+    this.dataSource.data.forEach((snapshot) => this.selection.select(snapshot));
+    this.cdr.markForCheck();
+  }
+
+  doAdd(): void {
+    this.modalService.openInSlideIn(SnapshotAddComponent);
+  }
+
+  doClone(snapshot: SnapshotListRow): void {
+    this.matDialog.open(SnapshotCloneDialogComponent, {
+      data: snapshot.id,
+    });
+  }
+
+  doRollback(row: SnapshotListRow): void {
```

Comment:
You can move the logic of querying for additional snapshot info to SnapshotRollbackDialog itself.